src/routes/login/+page.svelte

```diff
@@ -1,22 +1,115 @@
 <script>
+    // Framework imports
+    import { getToastStore } from "@skeletonlabs/skeleton";
+
+    // Import 🐦
+    import { ConstantBackoff, handleAll, retry } from "cockatiel";
+
+    // Import backend urls
+    import { urls } from "$lib/util.js";
+
     // Definitions
-    let somethingWrong = false;
+    // Constants
+    const toastStore = getToastStore();
+
+    const mailRegexPattern =
+        /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+
+    // Enums
+    const btnIcon = {
+        locked: "🔐",
+        unlocked: "🔓",
+    };
 
     const indicatorStatus = {
-        none: "",
+        none: "border-surface-700",
         sucess: "input-success",
-        warning: "input-warning",
-        error: "input-error",
+        warning: "border-warning-400",
+        error: "border-error-600",
+    };
+
+    // Policy for fetching
+    const retryPolicy = retry(handleAll, {
+        maxAttempts: 3, // Try 3 times
+        backoff: new ConstantBackoff(50), // Wait 10ms after each try
+    });
+
+    // Toast Settings
+    const toast = {
+        message: "Sie wurden erfolgreich angemeldet",
+        hideDismiss: true, // Hide the dismiss button on toast
+        timeout: 3000, // Auto dismiss toast after 3 seconds
+        background: "variant-filled-secondary",
     };
-    // Will get value "input-error" or "input-warning" according to status of backend fetch and validators
+
+    // Variables
+    // Will get value according to status of backend fetch and validators
     let emailIndicator = indicatorStatus.none;
     let passwordIndicator = indicatorStatus.none;
 
+    // Formfield binding references
+    let emailReference;
+    let passwordReference;
+
+    let somethingWrong = false;
     let showPassword = false;
-    let btnIcon = {
-        locked: "🔐",
-        unlocked: "🔓",
-    };
+
+    let email = "";
+    let password = "";
+    let error = "";
+
+    // Functions
+    function validateEmail(email) {
+        return mailRegexPattern.test(email);
+    }
+
+    function resetIndicators() {
+        emailIndicator = indicatorStatus.none;
+        passwordIndicator = indicatorStatus.none;
+        somethingWrong = false;
+    }
+
+    function tryLogin() {
+        resetIndicators();
+        email = emailReference.value;
+        password = passwordReference.value;
+        if (validateEmail(email)) {
+            fetchLogin();
+        } else {
+            emailIndicator = indicatorStatus.warning;
+        }
+    }
+
+    async function fetchLogin() {
+        try {
+            const mybody = {
+                email: email,
+                password: password,
+            };
```

Comment:
Email und Passwort würde ich einheitlich über Parameterübergabe machen. Ist auch etwas einfacher zu lesen, da für mich erst unklar war was die beiden Variablen `email` und `password` genau machen.
```suggestion
    let error = "";

    // Functions
    function validateEmail(email) {
        return mailRegexPattern.test(email);
    }

    function resetIndicators() {
        emailIndicator = indicatorStatus.none;
        passwordIndicator = indicatorStatus.none;
        somethingWrong = false;
    }

    function tryLogin() {
        resetIndicators();
        let email = emailReference.value;
        let password = passwordReference.value;
        if (validateEmail(email)) {
            fetchLogin(email, password);
        } else {
            emailIndicator = indicatorStatus.warning;
        }
    }

    async function fetchLogin(email, password) {
        try {
            const mybody = {
                email: email,
                password: password,
            };
```

Reply:
Leider stellt Github die Änderungen nicht so gut dar, deswegen hier noch ein Bild
![grafik](https://github.com/solid-stack-solutions/voycar-frontend/assets/58894308/5e8e2344-eea6-45f1-8ee2-4589937bb945)